### DISCUSSION TITLE
Add support for tenant members

### DISCUSSION THREAD

**Discussion by ioquatix** [auth0/terraform-provider-auth0#1176] [General]
(Copied from https://github.com/alexkappa/terraform-provider-auth0/issues/440)

Would be nice to be able to configure auth0 tenant members, currently have to add admins to every tenant individually by going through the settings

New or Affected Resource(s)

- `auth0_member` / `auth0_tenant`

## Potential Terraform Configuration

```
resource "auth0_member" "member" {
	email = "hello@example.com"
	role = "admin"
}
```

or something like

```
resource "auth0_tenant" "tenant" {
	member {
		email = "hello@example.com"
		role = "admin"
	}
}
```

**Comment by sergiught** [comment]
Hey @ioquatix, thanks for raising this issue with us. 

To set some expectations on this, as of right now the management API that this provider relies upon does not offer the possibility to manage tenant members. However we'll reach out internally and start a conversation around the possibility of offering this in the future. 

I'll circle back whenever I have more information on this topic. 

Appreciate your patience!

**Comment by ioquatix** [comment]
@sergiught Thanks for the update.

We saw from the logs that an API does appear to exist for this, it just seems to be undocumented. Do you know any more details about this? Basically do some operations in the tenant members list and you can see the log file showing an API being used.

**Comment by sergiught** [comment]
@ioquatix Indeed there is an API endpoint for managing tenant members that the Management Dashboard leverages, however this is still private and won't work with the client credentials flow that the terraform provider is using. It can only be called by a JWT Token issued to an already tenant admin member.

**Comment by ioquatix** [comment]
@sergiught thanks for the update.

We ended up writing selenium scripts which drive a web browser to provision, list and de-provision tenant members - we look forward to an Auth0 API + Terraform support in the future.

**Comment by papaharry** [comment]
Hello, just wanted to check if there was any update on this.
We currently have to add tenant members manually and we are thinking of writing a script to automate the provisioning/deprovisioning of members.

Should we go ahead with the script, or is this feature coming soon?

Thanks!

**Comment by sergiught** [comment]
Hey @papaharry 👋🏻 , unfortunately we don't have this on our radar at least for this quarter as the management API endpoints for this are not publicly accessible yet. As soon as they become available we'll add support for this.

**Comment by ioquatix** [comment]
@papaharry we wrote a script to do this and we might be able to share it. Let me check internally.

**Comment by hamza3202** [comment]
@ioquatix is it possible for you to share this script?

**Comment by ioquatix** [comment]
Unfortunately we are not able to share this code at this time.

**Comment by kenkoooo** [comment]
Any update?

**Comment by joewragg** [comment]
Would be a nice feature +1

**Comment by jcavet** [comment]
This is a no-brainer add, can we have someone give us an update?

**Comment by rajdotio** [comment]
Joining the queue. This is definitely a must have feature. I hope Auth0 adds it sooner. At least a GET endpoint that returns list of tenant members and their roles.
